src/controllers/controller.hpp

```diff
@@ -522,12 +522,23 @@ Result Controller::execute(std::vector<Circuit> &circuits,
     if (parallel_shots_ > 1 || parallel_state_update_ > 1)
       omp_set_nested(1);
 #endif
-    #pragma omp parallel for if (parallel_experiments_ > 1) num_threads(parallel_experiments_)
-    for (int j = 0; j < result.results.size(); ++j) {
-      // Make a copy of the noise model for each circuit execution
-      // so that it can be modified if required
-      auto circ_noise_model = noise_model;
-      execute_circuit(circuits[j], circ_noise_model, config, result.results[j]);
+    // then- and else-blocks have intentionally duplication.
+    // Nested omp has significant overheads even though a guard condition exists.
+    if (parallel_experiments_ > 1) {
+      #pragma omp parallel for if (parallel_experiments_ > 1) num_threads(parallel_experiments_)
```

Comment:
Do we still need the `if (parallel_experiments_ > 1)` within the `pragma`? Wouldn't this suffice now:
```
#pragma omp parallel for num_threads(parallel_experiments_)
```

Reply:
We need this `if`. Even the guard `parallel_experiments_ > 1` exists in pragma, OMP recognizes it is nested inside the loop body.

Reply:
So we need both `if`s?  In the first branch in the `pragma` we will never have `parallel_experiments_` <= 1,  right? Or am I missing something? Before #796 the code was:
```
if (parallel_experiments_ > 1) {
      // Parallel circuit execution
      #pragma omp parallel for num_threads(parallel_experiments_)
```
without `if` in the `pragma` statement. 
Sorry to insist, just want to be sure.

Reply:
sorry. You are right. `if (parallel_experiments_ > 1)` is not necessary in pragma. I will remove it.